### ISSUE TITLE
cl_downloadfilter causing inconsistency and ERRORs for modded servers

### ISSUE THREAD

**Issue by dragokas** [ValveSoftware/Source-1-Games#3429]
Since one of the latest updates Valve made a significant regression in relation to all L4D2 modded servers, where they become looking like a shit, with a lot of red ERRORs. Example:

![c8m2_subway0000](https://user-images.githubusercontent.com/19956568/97809393-45a31880-1c75-11eb-9354-2b2685b534b0.jpg)

These are the consequences of changing the default value "all" of a variable "cl_downloadfilter" to "none".
For 10 years it did not create problems for anyone.

This also creates inconsistency, where all old players have predefined/saved "cl_downloadfilter" "all" in their config.cfg file/cloud (have a priority), automatically created by the game. At the same time, all new players (who bought the game after the update) have "none" value. So, new players see "ERROR", old players don't.

Can you please, revert back this change, and return the default value of "cl_downloadfilter" to "all" ?
Thank you.

**Comment by MartLegion** [comment]
Agreed with Dragokas,

Not only new players have this issue,

Old players that reinstall the game (or install in another machine) and never had to activate it before, suffers from the same problem sometimes.

Of course, is a simple thing to do in the options, but is not practical.

I don't know why this got changed in the latest update, probably if it was modified to prevent some "exploit", was done in the wrong place.

Is almost the same as subscribing to an addon in the workshop, and days after, the author adds some malicious code in the .vpk

So please, if possible, rollback this var default to "all"

**Comment by 1157** [comment]
This does not make any sense. there are three options.When you get the "ERROR" a model is not packed in map or not downloaded, why would valve set to "NONE".Most dedicated servers run plug-ins to kick such "no-dl" clients.
http://risserver.ddns.net/forum/resources/download-filter-checker.125/

 Also do you have a PG rated screenshot rather than the one posted,(lol).

cl_downloadfilter all
Allows all files to be downloaded.

cl_downloadfilter nosounds
Doesn't allow sounds to be downloaded.

cl_downloadfilter none
Allows nothing to be downloaded.

**Comment by MartLegion** [comment]
Even after downloading the error model can still appear.
L4D/L4D2 works a bit differently than counter strike.
Also kicking clients is very ineffective. Most simply abandon the server.
The OP objective is to show how changing a default var that has been "all" for 10 years, can cause a lot of issues for running servers.
Mainly now in the last months of the year that we have remarkable events happening: "Halloween"/"Christmas"/"New Years".

**Comment by dragokas** [comment]
@1157, I'm not happy with the idea to kick everybody who have "none" value.
Because most of new players now have "none" value by default. That's mean - kick almost everybody and host empty server. lol
L4D/L4D2 has no such large community like CS or so.

1157, I don't understand your rofl. What is mean "PG rated screenshot" ?

**Comment by 1157** [comment]
Yes you cant change value mid-stream. if it is not in the cache you will see error. cl_downloadfiler is the same for Counter Strike ,l4d1/2.hl2mp,all Source Engine ect all the same filter a few games may have map only option.

I mentioned the kick plugin as this valve change will create new problems for players with that setting (now default) but it does give info to change setting. but with a lobby and the way l4d starts that's a hassle.

Valve needs to change setting back. This recent LS update may be a big problem Valve likes to start updates, break stuff, then never come back to fix.

As for the "pg" rating look a the nickname in your pic    IF**KYOUNOW

**Comment by Psykotikism** [comment]
As a 10-year veteran of this game, I agree with the petition to change the default value back to "all" in favor of community servers.

Everyone above me has already stated valid reasons for how this affects community servers, and so I will instead provide some insight into how this is problematic from an ordinary (no technical/programming knowledge) player's perspective.

As someone who doesn't host his own public server, I like to join community servers with my friends. Without knowing that the default value of the "cl_downloadfilter" setting has been changed from "all" to "none", my friends and I have been stumbling across several community servers with giant ERROR models everywhere. I was already aware of the change made to "cl_downloadfilter" so it was easy to fix on my end. My friends, however, didn't even know what that setting is for, so they just kept leaving all those servers out of frustration and annoyance. Now, if that setting's default value were reverted back to "all", people with no technical/programming experience like my friends wouldn't abandon community servers due to frustration/incompetence.

If the default value will not be reverted back to "all", then at least provide an option to toggle it in the main menu as Team Fortress 2 has. In TF2, all players can just go to the Multiplayer tab in their Options window and toggle their download filter. Since L4D2 doesn't have this tab/option available and I doubt adding it will be optimal, the best solution is to revert the default value.

**Comment by dragokas** [comment]
We have such an option under Network Settings -> Downloadable content.
However, again, it requires some technical knowledge to know where to find and why.

**Comment by rushianN1** [comment]
waiting update
